### PR TITLE
Playlist page

### PR DESCRIPTION
**Pull request by mikesch803**
1. 404 page
2. add playlist page (user can add/delete the playlist)
3. add modal to add video/remove from playlist in video page
4.  user can remove a video from playlist page

# files to check

- [ ] src/components/playlist-modal/PlaylistModal.jsx
- [ ] src/components/stack-list-container/StackListContainer.jsx
- [ ] src/context/playlist-context.jsx
- [ ] src/pages/404/NotFound.jsx
- [ ] src/pages/playlist-page/Playlist.jsx
- [ ] src/pages/video-page/Video.jsx
- [ ] src/reducer/PlaylistReducer.jsx

## test login
email : adarshbalika@gmail.com
 password : adarshBalika123

### CODE REVIEW

File: src/pages/playlist-page/Playlist.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    "header header header header"
+    "aside main main main";
+}
+
+.playlist-main {
+  grid-area: main;
+  margin: 1rem;
+}
+
+.flex-1 {
```
Comment (ApurvaSawant11) [inline src/pages/playlist-page/Playlist.css:14]: Add these utils in `App.css` instead

File: src/pages/playlist-page/Playlist.css [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  border: 1px solid;
+  background-color: white;
+  width: fit-content;
+  gap: 1px;
+  padding: 0.5rem;
+}
+
+.modal-playlist > p {
+  display: flex;
+  /* margin: 0; */
```
Comment (ApurvaSawant11) [inline src/pages/playlist-page/Playlist.css:52]: Remove this if not needed

File: src/pages/video-page/Video.css [hunk 3]
```diff
@@ -43,4 +44,20 @@
...
+  left: 0;
+  right: 0;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+}
+
+.blur {
+  background-color: rgba(200, 200, 200);
```
Comment (ApurvaSawant11) [inline src/pages/video-page/Video.css:62]: Try to use variables for reusability

File: src/pages/video-page/Video.jsx [hunk 4]
```diff
@@ -1,8 +1,13 @@
 import React from "react";
 import { useContext } from "react";
 import { Aside, VideoCard } from "../../components";
-import { VideoContext, WatchLaterContext } from "../../context";
+import { PlaylistModal } from "../../components/playlist-modal/PlaylistModal";
+import {
+  VideoContext,
+  WatchLaterContext,
+} from "../../context";
 import { LikedVideoContext } from "../../context/like-video-context";
```
Comment (ApurvaSawant11) [inline src/pages/video-page/Video.jsx:9]: You can add this import to the above context imports. Same for the `PlaylistModal` import and `useContext` as well. You are importing everything on different lines. Merge them

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me!

Just go through the suggestions

File: src/pages/playlist-page/Playlist.css [hunk 5]
```diff
@@ -0,0 +1,100 @@
+.playlist-grid-layout {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
```
Comment (Dev-AkshatGupta) [inline src/pages/playlist-page/Playlist.css:3]: Instead of writing again and again you can use repeat.

**Review by Dev-AkshatGupta — APPROVED** [review]
Everything else seems quite good to me nothing to suggest as everything else been suggested by Apurva

**Pull request merged**